src/sentry/spans/consumers/process/factory.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import annotations
+
+import uuid
+from datetime import datetime
+from typing import Any, Mapping, MutableMapping
+
+import msgpack
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.processing.strategies.abstract import ProcessingStrategy, ProcessingStrategyFactory
+from arroyo.processing.strategies.commit import CommitOffsets
+from arroyo.processing.strategies.produce import Produce
+from arroyo.types import Commit, Message, Partition, Topic
+from confluent_kafka import Producer
+
+from sentry import quotas
+from sentry.models import Project
+from sentry.spans.span import SPAN_SCHEMA_VERSION, Span
+from sentry.utils import json, kafka_config
+from sentry.utils.arroyo import RunTaskWithMultiprocessing
+
+
+def _build_snuba_payload(payload: Mapping[str, Any]) -> MutableMapping[str, Any]:
+    project = Project.objects.get_from_cache(id=payload["project_id"])
+    organization = project.organization
+
+    span_dict: MutableMapping[str, Any] = {}
+    span_dict["description"] = payload["description"]
+    span_dict["exclusive_time_ms"] = int(payload.get("exclusive_time", 0))
+    span_dict["is_segment"] = not payload.get("parent_span_id")
+    span_dict["organization_id"] = organization.id
+    span_dict["parent_span_id"] = payload.get("parent_span_id", "0")
+    span_dict["project_id"] = payload["project_id"]
+    span_dict["retention_days"] = quotas.get_event_retention(organization=organization) or 90
+    span_dict["span_id"] = payload.get("span_id")
+    span_dict["span_status"] = payload["status"]
+    span_dict["tags"] = payload.get("tags")
+    span_dict["trace_id"] = str(uuid.UUID(payload["trace_id"]))
+    span_dict["version"] = SPAN_SCHEMA_VERSION
+
+    start_timestamp = datetime.utcfromtimestamp(payload["start_timestamp"])
+    end_timestamp = datetime.utcfromtimestamp(payload["timestamp"])
+    span_dict["duration_ms"] = max(int((end_timestamp - start_timestamp).total_seconds() * 1e3), 0)
+
+    trace_context = payload["data"]["trace"]
+    span_dict["group_raw"] = trace_context["hash"]
+
+    sentry_tags: MutableMapping[str, Any] = {}
```

Comment:
I believe [these](https://github.com/getsentry/relay/blob/master/relay-dynamic-config/src/defaults.rs#L54-L67) would be the tags in the `sentry_tags` field. It would be ideal if the protocol from Relay to Ingest-Spans encoded this as well. That way no special handling would be needed when newer tags are extracted by Relay.